connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectMetrics.java

```diff
@@ -75,10 +78,17 @@ public ConnectMetrics(String workerId, WorkerConfig config, Time time) {
         MetricConfig metricConfig = new MetricConfig().samples(numSamples)
                 .timeWindow(sampleWindowMs, TimeUnit.MILLISECONDS).recordLevel(
                         Sensor.RecordingLevel.forName(metricsRecordingLevel));
-        JmxReporter jmxReporter = new JmxReporter(JMX_PREFIX);
+        JmxReporter jmxReporter = new JmxReporter();
         jmxReporter.configure(config.originals());
         reporters.add(jmxReporter);
-        this.metrics = new Metrics(metricConfig, reporters, time);
+        MetricsContext metricsContext = new KafkaMetricsContext(JMX_PREFIX, config.originals());
+        metricsContext.metadata().put(ConnectUtils.CONNECT_KAFKA_CLUSTER_ID, clusterId);
+        if (config.originals().get(CommonClientConfigs.GROUP_ID_CONFIG) != null) {
+            metricsContext.metadata().put(ConnectUtils.CONNECT_GROUP_ID,
+                    config.originals().get(CommonClientConfigs.GROUP_ID_CONFIG).toString());
```

Comment:
instead of repeating `config.originals().get(CommonClientConfigs.GROUP_ID_CONFIG)` here, you might want to extract the value into a variable before doing the check